src/internal/pfsdb/commits.go

```diff
@@ -502,10 +504,11 @@ func GetCommitAncestry(ctx context.Context, extCtx sqlx.ExtContext, startId Comm
 }
 
 // ForEachCommitAncestor queries postgres for ancestors of startId up to the maxDepth. cb() is called for each ancestor.
-// maxDepth is optional.
+// maxDepth is optional, but cannot exceed MaxSearchDepth. The caller may gracefully terminate iteration early by
+// returning errutil.ErrBreak in cb().
 func ForEachCommitAncestor(ctx context.Context, extCtx sqlx.ExtContext, startId CommitID, maxDepth uint, cb func(parentId, childId CommitID) error) error {
```

Comment:
If you like this call back pattern you can keep it, but if you write to the gRPC client with rows object open then the client can take up one of our precious Postgres connections for as long as they want.

Not a hard blocker, but if we call this from a gRPC context in a follow up PR, I would consider that a problem.

Reply:
That's fair. I don't foresee these being used in gRPC streams directly. I think for that, we would employ the iterator pattern that buffers a set of entries and then retrieves the next batch when its consumed all the entries.